### ISSUE TITLE
OpenSSL 3.0 uses different version scheme

### ISSUE THREAD

**Issue by 9cd2bb5e-c96d-422d-b12d-76bbef105a33** [python/cpython#90272]
BPO | [46114](https://bugs.python.org/issue46114)
--- | :---
Nosy | @tiran, @stratakis, @hroncok, @miss-islington, @penguinolog
PRs | <li>python/cpython#30170</li><li>python/cpython#30172</li><li>python/cpython#30173</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-12-17.10:51:19.891>
labels = ['expert-SSL', 'type-bug', '3.9', '3.10', '3.11', 'tests']
title = 'OpenSSL 3.0 uses different version scheme'
updated_at = <Date 2022-03-22.23:20:19.195>
user = 'https://github.com/penguinolog'
```

bugs.python.org fields:
```python
activity = <Date 2022-03-22.23:20:19.195>
actor = 'cstratak'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Tests', 'SSL']
creation = <Date 2021-12-17.10:51:19.891>
creator = 'penguinolog'
dependencies = []
files = []
hgrepos = []
issue_num = 46114
keywords = ['patch']
message_count = 12.0
messages = ['408770', '408773', '408776', '408779', '408788', '408792', '408795', '415499', '415502', '415504', '415517', '415839']
nosy_count = 5.0
nosy_names = ['christian.heimes', 'cstratak', 'hroncok', 'miss-islington', 'penguinolog']
pr_nums = ['30170', '30172', '30173']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue46114'
versions = ['Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by vstinner** [comment]
I created a backport to 3.8: https://github.com/python/cpython/pull/92954

**Comment by kumaraditya303** [comment]
Fixed by #30170
